### DISCUSSION TITLE
Count entities

### DISCUSSION THREAD

**Discussion by gamecoder-nz** [skypjack/entt#1077] [Q&A]
Is there a way to count the number of entities that have a given set of components?

**Comment by skypjack** [comment]
It depends on the set actually. Single type views have a `size` method that returns the exact number of elements while multi-type views have a `size_hint` method that return the upper bound. The only way to know the exact number in the last case is by visiting the set instead.

**Comment by gamecoder-nz** [comment]
Ok so I would need to keep track of the count myself somehow

**Reply by skypjack** [reply]
It's hard to say. It smells of XY-problem to be honest. What's the purpose? I really can't help without a bit of context.

**Comment by gamecoder-nz** [comment]
So I have a star spawner system. If the number of stars in the game is less
than 5 then create a new one. If there are already 5 stars then move the
oldest one.  In the collision detection system the star will be destroyed
if the player touches it.  I want to keep the systems separate and not
communicate with each other.

I originally had the size method because I was only finding entities with a
star component but I also need to exclude another component which means a
lose the size method

**Reply by skypjack** [reply]
Well it depends, if the second component is assigned only to _stars_ you have technically transformed the problem into a subtraction. 🤷‍♂️
